src/cockpit/polkit.py

```diff
@@ -0,0 +1,139 @@
+import asyncio
+import locale
+import logging
+import os
+import pwd
+
+from cockpit._vendor.systemd_ctypes import bus
+
+# that path is valid on at least Debian, Fedora/RHEL, and Arch
+HELPER_PATH = '/usr/lib/polkit-1/polkit-agent-helper-1'
+
+logger = logging.getLogger(__name__)
+
+
+# https://www.freedesktop.org/software/polkit/docs/latest/eggdbus-interface-org.freedesktop.PolicyKit1.AuthenticationAgent.html
+class org_freedesktop_PolicyKit1_AuthenticationAgent(bus.Object):
+    def __init__(self, responder):
+        super().__init__()
+        self.responder = responder
+
+    @bus.Interface.Method('', ['s', 's', 's', 'a{ss}', 's', 'a(sa{sv})'])
+    async def begin_authentication(self, action_id, message, icon_name, details, cookie, identities):
+        logger.debug('BeginAuthentication: %s %s %s %s %s %s',
+                     action_id, message, icon_name, details, cookie, identities)
+        # only support authentication as ourselves
+        my_uid = os.geteuid()
+        for (auth_type, subject) in identities:
+            if auth_type == 'unix-user' and 'uid' in subject and subject['uid'].value == my_uid:
+                logger.debug('Authentication subject %s matches our uid %d', subject, my_uid)
+                break
+        else:
+            logger.warning('Not supporting authentication as any of %s', identities)
```

Comment:
Why not?

Reply:
We don't currently have a UI to select an admin identifier (neither with the C bridge). This would be nice for the future (cf. your recently filed "not an admin user" issue, I can't find it quickly), but for now I'd like to get this common case working.

Reply:
I added a comment.

Reply:
I can understand preferring to do it as the current user, but what about the case where the only possibility is to do it as admin?  Why wouldn't we try?

Reply:
Don't we?  At the end of the day, we present the user with exactly the prompt message that we give to `do_askpass()`, plus or minus some "sudo polish", no?

I'm happy to keep that feature for later if you'd prefer not to spend time on it.

Reply:
I guess you resolved this because you prefer not to spend time on it? :)